editor/store/selectors.js

```diff
@@ -1354,28 +1408,16 @@ function getItemsFromInserts( state, inserts, allowedBlockTypes, maximum = MAX_R
  * @return {Editor.InserterItem[]} Items that appear in the 'Recent' tab.
  */
 export function getFrecentInserterItems( state, allowedBlockTypes, maximum = MAX_RECENT_BLOCKS ) {
-	const calculateFrecency = ( time, count ) => {
-		if ( ! time ) {
-			return count;
-		}
-
-		const duration = Date.now() - time;
-		switch ( true ) {
-			case duration < 3600:
-				return count * 4;
-			case duration < ( 24 * 3600 ):
-				return count * 2;
-			case duration < ( 7 * 24 * 3600 ):
-				return count / 2;
-			default:
-				return count / 4;
-		}
-	};
+	deprecated( 'getFrecentInserterItems', {
+		version: '3.0',
+		alternative: 'getInserterItems',
+		plugin: 'Gutenberg',
+	} );
 
-	const sortedInserts = values( state.preferences.insertUsage )
-		.sort( ( a, b ) => calculateFrecency( b.time, b.count ) - calculateFrecency( a.time, a.count ) )
-		.map( ( { insert } ) => insert );
-	return getItemsFromInserts( state, sortedInserts, allowedBlockTypes, maximum );
+	const items = getInserterItems( state, allowedBlockTypes );
+	const usefulItems = items.filter( ( item ) => item.utility > 0 );
+	const sortedItems = orderBy( usefulItems, [ 'utility', 'frecency' ], [ 'desc', 'desc' ] );
```

Comment:
As mentioned, I think any reference to `'frecency'` should be annotated as it's not an extremely common concept in my experience.

Reply:
This function will be deleted in 3.0 so I won't add a comment here.